packages/cdktf/lib/complex-computed-list.ts

```diff
@@ -491,3 +491,77 @@ export class AnyMapList extends MapList {
     return new AnyMap(this, `[${index}]`);
   }
 }
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+export class StringListMap extends ComplexMap {
+  constructor(
+    protected terraformResource: IInterpolatingParent,
+    protected terraformAttribute: string
+  ) {
+    super(terraformResource, terraformAttribute);
+  }
+
+  public get(key: string) {
+    return Token.asList(
+      this.terraformResource.interpolationForAttribute(
+        `${this.terraformAttribute}[${key}]`
+      )
+    );
+  }
+}
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+export class NumberListMap extends ComplexMap {
+  constructor(
+    protected terraformResource: IInterpolatingParent,
+    protected terraformAttribute: string
+  ) {
+    super(terraformResource, terraformAttribute);
+  }
+
+  public get(key: string) {
+    return Token.asNumberList(
+      this.terraformResource.interpolationForAttribute(
+        `${this.terraformAttribute}[${key}]`
+      )
+    );
+  }
+}
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+export class BooleanListMap extends ComplexMap {
+  constructor(
+    protected terraformResource: IInterpolatingParent,
+    protected terraformAttribute: string
+  ) {
+    super(terraformResource, terraformAttribute);
+  }
+
+  public get(key: string) {
+    // This isn't fully supported
+    return Token.asList(
```

Comment:
Given that `asList()` returns a `string[]`, wouldn't returning e.g. an `IResolvable` (via `Token.asAny`) be a bit better?
I forgot how Boolean lists where modeled for inputs 😅 Did they take `IResolvable` and/or `IResolvable[]`?

Reply:
My first thought was: "at least it's a list", but yeah, `IResolvable` seems better.